src/main.rs

```
@@ -88,7 +88,10 @@ fn interp(prg: &[u8], map: Vec<usize>) {
                 }
             }
             ',' => {
-                cells[ptr] = unsafe { getchar() } as u8;
+                let v = unsafe { getchar() };
+                if v != -1 {
```

Good catch.

I assume `EOF` is `-1`? Should we use `libc::EOF`?

https://docs.rs/libc/0.2.46/libc/constant.EOF.html

Errors are ignored, should we panic?

---

Agreed on `EOF`. I'm not sure about errors. Do you have an opinion?

---

@vext01 ping

---

Why not panic? Otherwise the output is incorrect.

---

OK, the new commit does that. It's more verbose than I expected!